btc-types/src/u256.rs

```diff
@@ -0,0 +1,310 @@
+use std::ops::{Div, Not, Rem, Shl, Shr};
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
+pub struct U256(u128, u128);
+
+impl U256 {
+    const MAX: U256 = U256(
+        0xffff_ffff_ffff_ffff_ffff_ffff_ffff_ffff,
```

Comment:
I would like to avoid custom changes over the original implementation https://github.com/rust-bitcoin/rust-bitcoin/blob/d5149c1f5445b28e73a38b58e9a864b0709c3c78/bitcoin/src/pow.rs#L482